batch.py

```diff
@@ -0,0 +1,324 @@
+#!/usr/bin/env python3
+# pylint: disable=wrong-import-position,import-error
+
+"""
+Hail Batch workflow for the rare-variant association analysis, including:
+
+- get relevant variants around a gene and export genotypes as plink files
+- generate input files for association tests 
+- run association tests
+"""
+
+# import python modules # is there a specific order, rationale for grouping imports??
+import os
+import re
+
+import click
+import logging
+
+from cloudpathlib import AnyPath
+from cpg_utils import to_path
+from cpg_utils.hail_batch import (
+    dataset_path,
+    get_config,
+    init_batch,
+    output_path,
+    remote_tmpdir,
+)
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from limix.qc import quantile_gaussianize
+from numpy import eye, ones
+from pandas_plink import read_plink1_bin
+from scipy.stats import shapiro
+
+# qvalue imports
+import scipy as sp
+from scipy import interpolate
+
+# make gene loc dict import
+from csv import DictReader
+
+import hail as hl
+import hailtop.batch as hb
+from hail.methods import export_plink
+
+# use logging to print statements, display at info level
+logging.basicConfig(
+    format="%(levelname)s:%(message)s", level=logging.INFO
+)  # consider removing some print statements?
+
+DEFAULT_JOINT_CALL_MT = dataset_path("mt/v7.mt")
+DEFAULT_ANNOTATION_HT = dataset_path(
+    "tob_wgs_vep/104/vep104.3_GRCh38.ht"
+)  # atm VEP only
+
+CELLREGMAP_IMAGE = (
+    "australia-southeast1-docker.pkg.dev/cpg-common/images/cellregmap:0.0.3"
+)
```

Comment:
Is the always the same as the driver image that's passed to the `analysis-runner` invocation? If so, you can fetch this value from the config: `get_config()['workflow']['driver_image']`.

Reply:
yes it should be, thanks good to know!

Reply:
i have tried but am slightly unsure of the syntax?